src/Sarif/Writers/SarifLogger.cs

```diff
@@ -323,7 +323,7 @@ private int LogRule(IRule iRule)
                 ruleIndex = _ruleToIndexMap.Count;
                 _ruleToIndexMap[rule] = ruleIndex;
                 _run.Resources = _run.Resources ?? new Resources();
-                _run.Resources.Rules = _run.Resources.Rules ?? new List<Rule>();
+                _run.Resources.Rules = _run.Resources.Rules ?? new OrderSensitiveValueComparisonList<Rule>(Rule.ValueComparer);
```

Comment:
>OrderSensitiveValueComparisonList [](start = 67, length = 33)

All this change does is enable Resources.Rules.Contains to have value, not reference, semantics. It's not strictly required for this change. Moving forward, we should figure out how/whether we should utilize this class across the SDK.